lambda/translcore.ml

```diff
@@ -524,7 +529,7 @@ and transl_exp0 ~in_new_scope ~scopes e =
          let fn = lfunction ~kind:Curried
                             ~params:[Ident.create_local "param", Pgenval]
                             ~return:Pgenval
-                            ~attr:default_function_attribute
+                            ~attr:function_attribute_disallowing_arity_fusion
```

Comment:
It's unclear to me why this is right. But @lpw25, @stedolan, and I couldn't come up with a scenario where the choice matters. Maybe at least comment that the choice is irrelevant? Or maybe you have a reason that I didn't guess.

Reply:
I think the choice matters, but not very much. The choice determines whether `lazy (let f x y = x + y in f)` is compiled as a  1-arg function returning a 2-arg function or as a 3-arg function. The former is a little better. (This example doesn't hit the `Constant_or_function` case above because it's a let-binding, but also `transl_exp` simplifies the let-binding.)

I added a comment to this effect.